internal/services/appservice/static_web_app_resource.go

```diff
@@ -105,6 +109,23 @@ func (r StaticWebAppResource) Arguments() map[string]*pluginsdk.Schema {
 
 		"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
 
+		"repository_url": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+		},
+
+		"repository_token": {
+			Type:      pluginsdk.TypeString,
+			Optional:  true,
+			Sensitive: true,
+		},
+
+		"repository_branch": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
```

Comment:
The validation that we're doing below in the Create and Update can be replaced by setting `RequiredWith` in the schema, we can also add some string validation for `repository_token` and `repository_branch`
```suggestion
		"repository_url": {
			Type:         pluginsdk.TypeString,
			Optional:     true,
			ValidateFunc: validation.IsURLWithHTTPorHTTPS,
			RequiredWith: []string{"repository_token", "repository_branch"},
		},

		"repository_token": {
			Type:      pluginsdk.TypeString,
			Optional:  true,
			Sensitive: true,
			ValidateFunc: validation.StringIsNotEmpty,
			RequiredWith: []string{"repository_url", "repository_branch"},
		},

		"repository_branch": {
			Type:     pluginsdk.TypeString,
			Optional: true,
			ValidateFunc: validation.StringIsNotEmpty,
			RequiredWith: []string{"repository_url", "repository_token"},
		},
```